packages/x-date-pickers-pro/src/internal/hooks/useDesktopRangePicker/useDesktopRangePicker.tsx

```diff
@@ -138,52 +133,22 @@ export const useDesktopRangePicker = <
     textField: slots.textField,
     root: slots.fieldRoot,
     separator: slots.fieldSeparator,
-    ...(fieldProps.slots ?? uncapitalizeObjectKeys(fieldProps?.components)),
+    ...fieldProps.slots,
   };
 
-  const slotPropsFromFieldProps = fieldProps.slotProps ?? fieldProps.componentsProps;
   const slotPropsForField: BaseMultiInputFieldProps<DateRange<TDate>, unknown>['slotProps'] = {
-    ...slotPropsFromFieldProps,
+    ...fieldProps.slotProps,
     textField: (ownerState) => {
       const externalInputProps = resolveComponentProps(slotProps?.textField, ownerState);
-      const inputPropsPassedByField = resolveComponentProps(
-        slotPropsFromFieldProps?.textField,
-        ownerState,
-      );
-      const inputPropsPassedByPicker =
-        ownerState.position === 'start' ? fieldSlotProps.startInput : fieldSlotProps.endInput;
-
       return {
         ...externalInputProps,
-        ...inputPropsPassedByField,
-        ...inputPropsPassedByPicker,
-        inputProps: {
-          ...externalInputProps?.inputProps,
-          ...inputPropsPassedByField?.inputProps,
-        },
+        ...(ownerState.position === 'start' ? fieldSlotProps.startInput : fieldSlotProps.endInput),
       };
     },
     root: (ownerState) => {
       const externalRootProps = resolveComponentProps(slotProps?.fieldRoot, ownerState);
-      const rootPropsPassedByField = resolveComponentProps(
-        slotPropsFromFieldProps?.root,
-        ownerState,
-      );
       return {
         ...externalRootProps,
-        ...rootPropsPassedByField,
-        ...fieldSlotProps.root,
-      };
-    },
-    separator: (ownerState) => {
-      const externalSeparatorProps = resolveComponentProps(slotProps?.fieldSeparator, ownerState);
-      const separatorPropsPassedByField = resolveComponentProps(
-        slotPropsFromFieldProps?.separator,
-        ownerState,
-      );
-      return {
-        ...externalSeparatorProps,
-        ...separatorPropsPassedByField,
         ...fieldSlotProps.root,
```

Comment:
Nice little bug spotted, we were passing the props of the root to the separator.